ci-operator/step-registry/ipi/conf/external-lb/ipi-conf-external-lb-commands.sh

```diff
@@ -0,0 +1,138 @@
+
+## HAProxy config
+cat > ${haproxy_cfg} << EOF
+defaults
+  maxconn 20000
+  mode    tcp
+  log     /var/run/haproxy/haproxy-log.sock local0
+  option  dontlognull
+  retries 3
+  timeout http-request 10s
+  timeout queue        1m
+  timeout connect      10s
+  timeout client       86400s
+  timeout server       86400s
+  timeout tunnel       86400s
+
+frontend api-server
+    bind ${BASTION_IP}:6443
+    default_backend api-server
+
+frontend machine-config-server
+    bind ${BASTION_IP}:22623
+    default_backend machine-config-server
+
+frontend router-http
+    bind ${BASTION_IP}:80
+    default_backend router-http
+
+frontend router-https
+    bind ${BASTION_IP}:443
+    default_backend router-https
+
+backend api-server
+    option  httpchk GET /readyz HTTP/1.0
+    option  log-health-checks
+    balance roundrobin
+$( for ip in "${master_ips[@]}"
+  do
+    echo "    server $ip $ip:6443 weight 1 verify none check check-ssl inter 1s fall 2 rise 3"
+  done
+)
+
+backend machine-config-server
+    balance roundrobin
+$( for ip in "${master_ips[@]}"
+  do
+    echo "    server $ip $ip:22623 check"
+  done
+)
+
+backend router-http
+    balance source
+    mode tcp
+$( for ip in "${worker_ips[@]}"
+  do
+    echo "    server $ip $ip:80 check"
+  done
+)
+
+backend router-https
+    balance source
+    mode tcp
+$( for ip in "${worker_ips[@]}"
+  do
+    echo "    server $ip $ip:443 check"
+  done
+)
+EOF
+
+# scp haproxy.cfg to bastion host /tmp/haproxy.cfg
+scp -o UserKnownHostsFile=/dev/null -o IdentityFile="${SSH_PRIV_KEY_PATH}" -o StrictHostKeyChecking=no "${haproxy_cfg}" ${BASTION_SSH_USER}@${BASTION_IP}:${bastion_haproxy_cfg}
+
+# Reload /tmp/haproxy.cfg by restart haproxy.service in bastion host
+ssh -o UserKnownHostsFile=/dev/null -o IdentityFile="${SSH_PRIV_KEY_PATH}" -o StrictHostKeyChecking=no ${BASTION_SSH_USER}@${BASTION_IP} "sudo mkdir -p /etc/haproxy; sudo cp ${bastion_haproxy_cfg} /etc/haproxy/haproxy.cfg; sudo systemctl restart haproxy.service"
+
+cluster_name=${NAMESPACE}-${JOB_NAME_HASH}
+base_domain=$(<"${SHARED_DIR}"/basedomain.txt)
+cluster_domain="${cluster_name}.${base_domain}"
+hosted_zone_id=$(<"${SHARED_DIR}"/hosted-zone.txt)
+
+# Update DNS to use external lb ip
+echo "Updating DNS records..."
+cat > "${SHARED_DIR}"/dns-update.json <<EOF
+{
+"Comment": "Update public OpenShift DNS records to use external lb",
+"Changes": [{
+    "Action": "UPSERT",
+    "ResourceRecordSet": {
+      "Name": "api.$cluster_domain.",
+      "Type": "A",
+      "TTL": 60,
+      "ResourceRecords": [{"Value": "${BASTION_IP}"}]
+      }
+    },{
+    "Action": "UPSERT",
+    "ResourceRecordSet": {
+      "Name": "*.apps.$cluster_domain.",
+      "Type": "A",
+      "TTL": 60,
+      "ResourceRecords": [{"Value": "${BASTION_IP}"}]
+      }
+}]}
+EOF
+id=$(aws route53 change-resource-record-sets --hosted-zone-id "$hosted_zone_id" --change-batch file:///"${SHARED_DIR}"/dns-update.json --query '"ChangeInfo"."Id"' --output text)
```

Comment:
I did not see the step of feeding aws credentails to run the aws command in this script.